### PR TITLE
Do not use object destructuring in state because it breaks the minifier.

### LINKED ISSUE #2275: state branch: error in gijoe when clicking for the profile page
`Uncaught TypeError: Cannot read property 'length' of null` in generateState

### PR DESCRIPTION
**Pull request by hatched**
Fixes #2275

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

As long as the tests pass, it's ok.

File: jujugui/static/gui/src/app/state/state.js [hunk 1]
```diff
@@ -473,7 +473,14 @@ const State = class State {
     }
     // Extract out the user sections.
     if (parts.includes(PATH_DELIMETERS.get('user'))) {
-      ({state, parts, error} = this._parseUser(parts, state));
+      // XXX This return value isn't using object destructuring because babili
+      // minifier incorrectly munges the return values and it breaks. In the
+      // future it can be reverted and tested against a newer version than
+      // 0.0.9
+      let parsed = this._parseUser(parts, state);
```
Comment (frankban) [inline jujugui/static/gui/src/app/state/state.js]: const?

Comment (frankban) [inline jujugui/static/gui/src/app/state/state.js]: ah, it's already used before, nm.

Comment (frankban) [inline jujugui/static/gui/src/app/state/state.js]: no it's not, so change it.

**Comment by hatched** [comment]
Thanks @frankban :) :shipit:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6319/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6321/
Test PASSed.

**Pull request merged**